src/components/SharedComponents/SwitchComponent.tsx

```diff
@@ -5,40 +5,33 @@ import { SwitchGrid, SwitchLabel } from './SwitchComponent.style';
 const SwitchComponent = (props: {
   leftLabel: string;
   rightLabel: string;
-  checkedValue: boolean | undefined;
-  onChange: any;
-  gridOnClick: any;
+  checked: boolean | undefined;
+  onChange: (newValue: boolean) => void;
```

Comment:
Removed `gridOnClick` and changed `onChange` so it takes a boolean so it can fill the role.